### ISSUE TITLE
IServicePrincipalCredential error during setup

### ISSUE THREAD

**Issue by jerone** [microsoft/AzureStorageExplorer#4653]
### Preflight Checklist

- [X] I have installed the [latest version of Storage Explorer](https://github.com/Microsoft/AzureStorageExplorer/releases/latest).
- [X] I have checked existing resources, including the [troubleshooting guide](https://docs.microsoft.com/azure/storage/common/storage-explorer-troubleshooting) and the [release notes](https://github.com/microsoft/AzureStorageExplorer/releases).
- [X] I have searched for [similar issues](https://github.com/Microsoft/AzureStorageExplorer/issues).

### Storage Explorer Version

1.20.0

### Regression From

No

### Architecture

x64

### Storage Explorer Build Number

20210625.5

### Platform

Windows

### OS Version

Microsoft Windows 10 Enterprise Version	10.0.19043 Build 19043

### Bug Description

During setup of installing the latest version, an error is shown multiple times. 

![image](https://user-images.githubusercontent.com/55841/124458120-e873df80-dd8c-11eb-99f1-d1f5d87f16dd.png)

Retry doesn't help.

### Steps to Reproduce

1. Download [latest installer](https://azure.microsoft.com/en-us/features/storage-explorer/).
2. Run installer.

### Actual Experience

During setup of installing the latest version, an error is shown multiple times. 

![image](https://user-images.githubusercontent.com/55841/124458120-e873df80-dd8c-11eb-99f1-d1f5d87f16dd.png)

Retry doesn't help.

Skipping the file does work, and the application will open after installation. No visual errors.

Reinstalling gives the same error message.

### Expected Experience

To not have this experience during setup.

### Additional Context

I'm full admin of my environment.

**Comment by MRayermannMSFT** [comment]
@jerone huh that's weird. I don't think we've ever seen this before. Hmmm. I think skipping should be fine as well, especially since it sounds like everything is working well for you. Can you tell me how long the path for that file is (including the file)? My only guess would be that the path for it is too long and that's causing some sort of issue during install (but then how did it get installed in the first place :P).

If at any point the app starts misbehaving and you want to try getting a successful install you could try these ideas:
- Uninstall and then reinstall
- Restart computer and then try running the install again

But ya, if the app is behaving fine then I wouldn't worry about this for now....

**Comment by jerone** [comment]
> Can you tell me how long the path for that file is (including the file)? 

@MRayermannMSFT Can you tell me where I can copy this path from? Is there a log?

**Comment by MRayermannMSFT** [comment]
@jerone how about this, can you tell me how long the path is to the install folder for storage explorer?

**Comment by jerone** [comment]
> @jerone how about this, can you tell me how long the path is to the install folder for storage explorer?

Sorry for the late reply:

```
C:\Users\<18 chars name>\AppData\Local\Programs\Microsoft Azure Storage Explorer
```

**Comment by thorhj** [comment]
I am getting the same error. If I ignore the error I get a similar error for `IStorageCredential.js` (and `.d.ts`). The application itself seems to be working by ignoring the errors (using AAD authentication).

My name in Windows is 20 characters long if that helps.

**Comment by MRayermannMSFT** [comment]
Without having done the math, I did do a quick search for the error code, and "code 3" does indicate a path length issue.

**Comment by MRayermannMSFT** [comment]
Relative to the root of the install directory, our longest path length went from 147 chars in 1.19.1, to 178 chars in 1.20.0. So for @jerone, whose install directory path is 85 characters, that's 263 total, which is 3 past the standard Windows limit of 260. I think likely we're going to do something about this.

**Comment by MRayermannMSFT** [comment]
We're going to hotfix this issue. For now though, if someone runs into this when running the installer for 1.20.0, it is probably safe to just skip. Alternatively, you can install in a different location, or if you are upgrading, uninstall using the uninstall.exe, and then reinstall in a different location.

**Comment by MRayermannMSFT** [comment]
Hey @jerone @thorhj @SharmaHarsh7 if y'all have a moment, it'd be great if you could install this 1.20.1 release candidate and let us know if you still see the install error. Thanks!

https://storageexplorerpublish.blob.core.windows.net/privatebuilds/1.20.1-RC1/StorageExplorer-ia32.exe

**Comment by thorhj** [comment]
@MRayermannMSFT the shady .exe file you posted here on Github was download and installed without issues for me 💪

**Comment by jerone** [comment]
@MRayermannMSFT Fixed for me too :+1:

**Comment by MRayermannMSFT** [comment]
Hey all, thanks for the help with this issue. We've now shipped 1.20.1. This build is almost the same as what I shared with you, except it includes 1 additional bug fix. I encourage y'all to go ahead and download the official 1.20.1 release from https://storageexplorer.com. Unfortunately y'all won't get an update notification to that version, but you will eventually get an update notification to whatever our next version is (e.g. 1.21.0). Thanks again and always feel free to share feedback and other issues! :)
